coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/PlaybackToSecondary.scala

```diff
@@ -265,6 +265,26 @@ class PlaybackToSecondary[CT, CV](u: PlaybackToSecondary.SuperUniverse[CT, CV],
       retrying[T](actions, filter)
     }
 
+    /**
+      * Serialize resync in the same store group
+      * because resync will cause reads to be unavailable (at least in pg)
+      */
+    def resyncSerially(): Unit = {
+      val sm = u.secondaryManifest
+      try {
+        // resync immediately calls commit before actual work is done.
+        // That allows us to see resync in progress from the resync table.
+        if (sm.lockResync(datasetId, secondary.storeId, secondary.groupName) > 0) {
+          resync()
```

Comment:
so if there is already a lock... what happens exactly? I don't think the the worker will release it's claim properly. I think that needs to be pulled out of resync() and put here.